repos/system_upgrade/common/libraries/overlaygen.py

```diff
@@ -228,3 +505,119 @@ def create_source_overlay(mounts_dir, scratch_dir, xfs_info, storage_info, mount
     except Exception:
         cleanup_scratch(scratch_dir, mounts_dir)
         raise
+
+
+# #############################################################################
+# Deprecated OVL solution ...
+# This is going to be removed in future as the whole functionality is going to
+# be replaced by new one. The problem is that the new solution can potentially
+# negatively affect systems with many loop mountpoints, so let's keep this
+# as a workaround for now. I am separating the old and new code in this way
```

Comment:
I'd just add that it is an optional workaround triggered by the LEAPP_OVL_LEGACY env. variable. Just for the comment to contain the complete information if it stays there longer.

Reply:
covered